### PR TITLE
Adds crate animation

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

 Adjusts the code from the locker animation Port PR originally from monster860 https://github.com/BeeStation/BeeStation-Hornet/pull/2665 accordingly to animate crates as well as splits all crate sprites up into its base open closed and back sprite.
https://streamable.com/ph58uw

## Why It's Good For The Game

Adds animated crates just like animated lockers

## Changelog
:cl:
add: animated crates
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -91,20 +92,37 @@
...
-
-		if(I == 0)
-			door_obj.transform = M
-			door_obj.icon_state = door_state
-			door_obj.layer = door_layer
-		else if(I == 1)
-			animate(door_obj, transform = M, icon_state = door_state, layer = door_layer, time = world.tick_lag, flags = ANIMATION_END_NOW)
-		else
-			animate(transform = M, icon_state = door_state, layer = door_layer, time = world.tick_lag)
+	if(is_crate)
```
Comment (ike709) [inline code/game/objects/structures/crates_lockers/closets.dm]: Just override the proc on the crate subtype.

Comment (MNarath1) [reply code/game/objects/structures/crates_lockers/closets.dm]: yeah thats definitively better i tried that at the beginning but i think i messed something up initially and went with the if(crate) and just never changed it back

**Comment by Isy232** [comment]
Looks good

**Comment by qwertyquerty** [comment]
How did this take 33 commits

**Comment by MNarath1** [comment]
> How did this take 33 commits

me uploading commits to early to my own repro so yeah and then of course the fun when i forgot stuff

**Review by PowerfulBacon — APPROVED** [review]

**Comment by ike709** [comment]
...I assume the testmerge was fine?

**Comment by PowerfulBacon** [comment]
Oh forgot to merge

**Pull request merged**